### PR TITLE
feat: add `babel-plugin-tester` types

### PR DESCRIPTION
**Pull request by ifiokjr**
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` should be present and it shouldn't have any additional or disabling of rules. Just content as `{ "extends": "dtslint/dt.json" }`. If for reason the some rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]`  and not for whole package so that the need for disabling can be reviewed.
- [x] `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

**Comment by ifiokjr** [comment]
Resubmitting after #44510 was automatically closed.

File: types/babel-plugin-tester/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,293 @@
...
+     *     // your test objects
+     *   },
+     * });
+     * ```
+     *
+     */
+    babelOptions?: Babel.TransformOptions;
+}
+
+type Omit<T, K extends keyof any> = Pick<T, Exclude<keyof T, K>>;
```
Comment (peterblazejewicz) [inline types/babel-plugin-tester/index.d.ts:157]:
```suggestion
export type Omit<T, K extends keyof any> = Pick<T, Exclude<keyof T, K>>;
```
that's ok, and will remove that hack with `export {}`

File: types/babel-plugin-tester/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,293 @@
...
+ * Formatter used for the snapshots.
+ */
+export function prettierFormatter(code: string, options: Options): string;
+
+export const unstringSnapshotSerializer: {
+    test(value: unknown): value is string;
+    print(value: string): string;
+};
+
+export {};
```
Comment (peterblazejewicz) [inline types/babel-plugin-tester/index.d.ts:293]:
```suggestion
```

**Review by peterblazejewicz — APPROVED** [review]
LGTM!
@ifiokjr thx for the contribution!

File: types/babel-plugin-tester/index.d.ts (hunk 1 above)
Comment (ifiokjr) [reply types/babel-plugin-tester/index.d.ts:157]: The problem with that is that it overloads the `Omit` import for pre ts@3.5 projects. Auto completion for `Omit` or similar popular utilities shows multiple packages and it's often difficult to know which project the type is going to be imported from.

It's just something I've noticed.

**Review by peterblazejewicz — APPROVED** [review]
LGTM!
@ifiokjr thx for the contribution!

**Pull request merged**